src/components/screens/HomeCarousel.js

```diff
@@ -12,71 +12,58 @@ import { LoadingContainer } from '../LoadingContainer';
 import { getQuery, QUERY_TYPES } from '../../queries';
 import { graphqlFetchPolicy, parsedImageAspectRatio } from '../../helpers';
 import { useRefreshTime } from '../../hooks';
+import { useHomeRefresh } from '../../hooks/HomeRefresh';
 
-export const HomeCarousel = ({ navigation, refreshing }) => {
+export const HomeCarousel = ({ navigation }) => {
   const { isConnected, isMainserverUp } = useContext(NetworkContext);
   const { globalSettings } = useContext(SettingsContext);
 
   const refreshTime = useRefreshTime('publicJsonFile-homeCarousel');
 
-  if (!refreshTime) {
-    return (
-      <LoadingContainer>
-        <ActivityIndicator color={colors.accent} />
-      </LoadingContainer>
-    );
-  }
-
   const fetchPolicy = graphqlFetchPolicy({
     isConnected,
     isMainserverUp,
     refreshTime
   });
 
-  return (
-    <Query
-      query={getQuery(QUERY_TYPES.PUBLIC_JSON_FILE)}
-      variables={{ name: 'homeCarousel' }}
-      fetchPolicy={fetchPolicy}
-    >
-      {({ data, loading, refetch }) => {
-        // call the refetch method of Apollo after `refreshing` is given with `true`, which happens
-        // when pull to refresh is used in the parent component
-        if (refreshing) refetch();
-        if (loading) {
-          return (
-            <LoadingContainer>
-              <ActivityIndicator color={colors.accent} />
-            </LoadingContainer>
-          );
-        }
+  const { data, loading, refetch } = useQuery(getQuery(QUERY_TYPES.PUBLIC_JSON_FILE), {
+    variables: { name: 'homeCarousel' },
+    fetchPolicy,
+    skip: !refreshTime
```

Comment:
this `skip` is new here but i cannot find out why?!
"skip if there is no refresh time"? what case did you think of here?

Reply:
the case i thought of was "refresh time is undefined", which will happen on every initial render ;)

due to the fact how the refresh time is loaded from async storage, it is always undefined at first.

previously we had an early return of a loadingspinner, if the refreshTime was undefined, before the query was executed.

because we now use the hook (and the amount and order of hooks needs to remain consistent between renders) we need to execute the hook with the skip property instead, if the refreshTime is undefined.

Reply:
clear description i couldn't think of/find by myself with looking on the diff.
thanks. agree.

if we wouldn't skip, the query would request fresh data before knowing, that there is a cached version from within the refresh time, right? (if there is)

Reply:
exactly. the reasoning for the skip is exactly the same as we had before with the early return